packages/commerce-sdk-react/src/hooks/sample.test.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useEffect, useState} from 'react'
+import '@testing-library/jest-dom'
+import {screen, waitFor} from '@testing-library/react'
+
+import {mockHttpResponses, renderWithProviders} from '../test-utils'
+import useCommerceApi from './useCommerceApi'
+
+const {withMocks} = mockHttpResponses({directory: `${__dirname}/mock-responses`})
+
+// NOTE: how to easily _update_ the mocked responses:
+// - Globally, via `npm run test:update-mocked-responses`
+// - Per individual test file, by passing in `mode` value -> mockHttpResponses({mode: 'update', directory})
+
+const ProductName = ({id}: {id: string}) => {
+    const api = useCommerceApi()
+    const [productName, setProductName] = useState('')
+
+    useEffect(() => {
+        const fetchProductName = async () => {
+            const product = await api.shopperProducts.getProduct({
+                parameters: {id}
+            })
+            setProductName(product.name as string)
+        }
+        fetchProductName()
+    }, [])
+
+    return <div>{productName}</div>
+}
+
+test(
+    'useCommerceApi hook',
+    withMocks(async () => {
+        // This util function would automatically integrate the provider components for you
+        renderWithProviders(<ProductName id="25591862M" />)
```

Comment:
I like this. 

I know you don't have much implementation to work with, but maybe we can think of providing even more utility in the future so we can have the function create the component for us. Then it can be used with various hooks.

Things we need to test for each hook at that they return data when valid params are set, that is can render an error if one if set, and maybe the loading state? (not too sure about that one). 

I have some ideas, maybe we can talk about it. But in short is would be nice to have an array that can be processed that has different "test settings". 

```
consts tests = [{
    hook: getProduct,
    params: {id: VALID_PRODUCT_ID},
    assertion: async (screen) => {
        expect(screen.getById('object-json')).toBeInTheDocument()
    }
}, {
    hook: getProduct,
    params: {id: INVALID_PRODUCT_ID},
    assertion: async (screen) => {
        expect(screen.getById('object-error')).toBeInTheDocument()
    }
}]
```

Reply:
Oh I see. Right, there would be common patterns among the hooks tests that we can extract out.

Reply:
@bendvc would it be all right if we create additional utility functions in a separate PR and at a later time? The reasons are:
- I can't really test well my code, since there are no finished implementations/interfaces of `useAsync`, `useProduct` etc. hooks yet in my branch.
- I wonder if it's really necessary to test passing in valid and invalid arguments. That feels like we're testing the underlying isomorphic SDK. Can we assume that it's well-tested already? See my comments in this test for more details: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/682/files#diff-9f41e237425df4cd7226a2317ff002601db30e7034c03f4d8b2105caeefd2ee1R105